### ISSUE TITLE
Git branching for handling staggered releases

### ISSUE THREAD

**Issue by gotoplanb** [Mobiquity/team-stanton#8]
Talking with Rob Goff about how to update the git-flow branching model to better facilitate Build and Support work. Need to rethink http://nvie.com/posts/a-successful-git-branching-model/

Scenario: 

* prod bug found in `release/1.0` or could just be a change request
* Support creates `release/1.0.1` for the maintenance release with `release/1.0` as base
* create `support/1.0.1/JIRA-123` to complete the needed Support work
* open pr into `release/1.0.1` and open pr into `develop` as opposed to typical merge to `develop` then having to use a label or comment to remind to cherrypick into a release
* ? who deletes this support branch since there are two prs?

Other considerations:

* QA builds should either be manual trigger from `develop` or we should use `master` or `qa` branch name to indicate ready for QA testing. Goal is to keep partially complete work from getting premature QA, but we also want Build devs to merge into `develop` as soon as possible to mitigate unnecessary merge conflicts
* Jenkins RC builds should be started with parametized build config that explicitly states the release branch to use. This allows different countries to use different releases.

Rob:

Per your scenario:

The `support/1.0.1/JIRA-123` branch would probably be deleted by Build devs after merge to `develop`, or if they determined that code was not required. (For example, fix superseded by new/modified feature in new build project.)

Dave:

`patch` prefixing. JIRA cloned tickets for all different merge targets

**Comment by gotoplanb** [comment]
feature, bug, patch

**Comment by firoze** [comment]
how do you `tag the release commits` for a project's repository?

**Comment by gotoplanb** [comment]
From Rob related to a separate QA branch (which I presonally disagree with creating)

I think the reasoning was:
- Fred just checked in code to develop branch. Fred is a rock star, great code, few problems.
- Barney just checked in code to develop branch. Barney’s not as good as Fred.
- PM wants a build to demo to client at end of sprint.

Rob Goff [1:25 PM]
- QA has tested, Fred’s commit is A-OK, Barney’s commit has caused regression bugs.

**Comment by gotoplanb** [comment]
`git pull origin develop`

`git checkout -b release/2.0`

`git push origin release/2.0`

`git tag release/2.0.0`

`git push origin --tags`

Then go to GitHub and open a pull request from `release/2.0` to `master`

Once PR is merged, delete the release branch, since you've shipped this to production. If a hotfix branch emerges, you can re-created the branch from the tag you setup. We tag down to bug fix dot notation always for future proofing.
